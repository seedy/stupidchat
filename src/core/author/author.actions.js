export const CONNECT = 'core/author/CONNECT';

export function login(author) {
  return {type: CONNECT, payload: author};
}