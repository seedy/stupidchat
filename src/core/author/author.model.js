export class AuthorModel {
  /**
   *
   * @type {Object.<string, string>}
   */
  static tzs = {
    paris: 'Europe/Paris',
    chicago: 'America/Chicago'
  };

  /**
   *
   * @param city
   * @returns {string}
   */
  static cityToLocale(city) {
    return AuthorModel.tzs[city];
  }

  /**
   *
   * @param id string
   * @param name string
   * @param surname string
   * @param city string
   * @param locale string
   */
  constructor(id, name, surname, city, locale) {
    this.id = id;
    this.name = name;
    this.surname = surname;
    this.city = city;
    this.locale = locale;
  }

  /**
   *
   * @param author {AuthorModel|number}
   * @returns {boolean}
   */
  isSame(author) {
    if (author instanceof AuthorModel) {
      return this.id === author.id;
    }
    if (typeof author === 'number') {
      return this.id === author;
    }
    return false;
  }

  /**
   * @returns {string}
   */
  toString() {
    return this.name + ' ' + this.surname;
  }

  /**
   *
   * @returns {*}
   */
  getInitial() {
    return this.name[0];
  }
}