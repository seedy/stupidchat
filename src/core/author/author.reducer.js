import {CONNECT} from "./author.actions";


export function authorById(state = {}, action) {
  switch (action.type) {
    case CONNECT:
      const author = action.payload;
      return Object.assign({}, state, {
        [author.id]: author
      });
    default: return state;
  }
}