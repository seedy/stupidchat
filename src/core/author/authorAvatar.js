import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import red from '@material-ui/core/colors/red';
import {AuthorModel} from "./author.model";


const styles = {
  avatar: {
    height: '30px',
    width: '30px',
    backgroundColor: red[500],
    margin: '5px'
  }
};

class AuthorAvatar extends Component {
  constructor(props) {
    super(props)
  }


  render() {
    const {classes} = this.props;
    return (
      <Avatar aria-label={this.props.author.toString()} className={classes.avatar}>
        {this.props.author !== '' ? this.props.author.getInitial() : ''}
      </Avatar>
    )
  }
}

AuthorAvatar.propTypes = {
  author: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(AuthorModel)])
};
AuthorAvatar.defaultProps = {
  author: ''
};
export default withStyles(styles)(AuthorAvatar);