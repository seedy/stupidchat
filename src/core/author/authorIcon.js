import React, {Component} from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';
import Tooltip from '@material-ui/core/Tooltip';
import {AuthorModel} from "./author.model";

export class AuthorIcon extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Tooltip title={this.props.author.toString()}>
        <IconButton
          aria-label={this.props.author.toString()}
          color="inherit"
        >
          <Icon>account_circle</Icon>
        </IconButton>
      </Tooltip>
    )
  }
}

AuthorIcon.propTypes = {
  author: PropTypes.oneOfType([PropTypes.instanceOf(AuthorModel), PropTypes.string])
};
AuthorIcon.defaultProps = {
  author: ''
};
