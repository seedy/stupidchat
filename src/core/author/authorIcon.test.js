import React from 'react';

import renderer from 'react-test-renderer';
import {AuthorIcon} from "./authorIcon";
import {AuthorModel} from "./author.model";
import {TestHelper} from "../test.helper";

describe('testing author icon', () => {
  const context = {};
  const author = new AuthorModel(1, 'John', 'Doe');

  beforeEach(() => {
    context.author = author;
  });

  afterEach(() => {
    TestHelper.clearContext(context);
  });

  test('AuthorIcon', () => {
    const spy = jest.spyOn(context.author, 'toString');
    const component = renderer.create(
      <AuthorIcon author={context.author} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(spy.mock.calls.length).toBe(2);

    spy.mockRestore();
  });
});
