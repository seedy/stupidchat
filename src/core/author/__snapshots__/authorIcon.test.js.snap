// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing author icon AuthorIcon 1`] = `
<button
  aria-describedby={null}
  aria-label="John Doe"
  className="MuiButtonBase-root-14 MuiIconButton-root-8 MuiIconButton-colorInherit-9"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  title="John Doe"
  type="button"
>
  <span
    className="MuiIconButton-label-13"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-17"
    >
      account_circle
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-26"
  />
</button>
`;
